btsieve/src/bitcoin/blockchain_info_hex_bitcoin_http_blocksource.rs

```diff
@@ -0,0 +1,169 @@
+    BlockDeserialization(String),
+}
+
+#[derive(Clone)]
+pub struct BlockchainInfoHexHttpBlockSource {
+    network: Network,
+    client: Client,
+}
+
+impl BlockchainInfoHexHttpBlockSource {
+    pub fn new(network: Network) -> Result<Self, Error> {
+        // Currently configured for Mainnet only because no support for hex-encoded
+
+        match network {
+            Network::Mainnet => Ok(Self {
+                network,
+                client: Client::new(),
+            }),
+            _ => {
+                log::error!(
+                    "Network {} not supported for bitcoin http blocksource",
+                    network
+                );
+                Err(Error::UnsupportedNetwork(format!(
+                    "Network {} currently not supported for bitcoin http plocksource",
+                    network
+                )))
+            }
+        }
+    }
+
+    pub fn latest_block(&self) -> impl Future<Item = MinedBlock, Error = Error> + Send + 'static {
+        let cloned_self = self.clone();
+
+        self.latest_block_without_tx()
+            .and_then(move |latest_block| {
+                cloned_self.raw_hex_block(latest_block.hash, latest_block.height)
+            })
+    }
+
+    fn base_url(&self) -> String {
+        match self.network {
+            Network::Mainnet => "https://blockchain.info".to_string(),
+            _ => panic!(
+                "Network {} not supported for bitcoin.info blocksource",
+                self.network
+            ),
+        }
+    }
+
+    fn latest_block_without_tx(
+        &self,
+    ) -> impl Future<Item = BlockchainInfoLatestBlock, Error = Error> + Send + 'static {
+        // https://blockchain.info/q/latesthash only works for mainnet, there is no testnet endpoint
+        // we fall-back to [testnet.]blockchain.info/latestblock to retrieve the latest
+        // block hash
+
+        let latest_block_url = format!("{}/latestblock", self.base_url());
+
+        self.client
+            .get(latest_block_url.as_str())
+            .send()
+            .map_err(Error::Reqwest)
+            .and_then(move |mut response| {
+                response
+                    .json::<BlockchainInfoLatestBlock>()
+                    .map_err(Error::Reqwest)
+            })
+    }
+
+    fn raw_hex_block(
+        &self,
+        block_hash: String,
+        block_height: u32,
+    ) -> impl Future<Item = MinedBlock, Error = Error> + Send + 'static {
+        let raw_block_by_hash_url =
+            format!("{}/rawblock/{}?format=hex", self.base_url(), block_hash);
+
+        self.client
+            .get(raw_block_by_hash_url.as_str())
+            .send()
+            .map_err(Error::Reqwest)
+            .and_then(|mut response| response.text().map_err(Error::Reqwest))
+            .and_then(|response_text| hex::decode(response_text).map_err(Error::Hex))
+            .and_then(move |bytes| {
+                deserialize(bytes.as_ref())
+                    .map(|block| {
```

Comment:
You (or I) could try to move the `.map` outside the `and_then` (aesthetics)

Reply:
Should work yes:

```rust
.and_then(move |bytes| deserialize(bytes.as_ref()).map_err(|e| ...))
.map(|block| MinedBlock::new(block, block_height))
```